### DISCUSSION TITLE
Computed property value based of other property

### DISCUSSION THREAD

**Discussion by vanvanni** [spatie/laravel-data#209] [Q&A] [answered]
Hi!

In my case I have a name but that name would be the id if it is slugified. So is their a way I can create a **id property** that is automatically computed of the **name property** with the Str::slug option?

**Comment by Nir-An** [comment, marked as answer]
Hi @gitdevries 

I think it could be local cast:

https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/casts#content-local-casts
https://spatie.be/docs/laravel-data/v2/advanced-usage/creating-a-cast

also, you can use transformers:

https://spatie.be/docs/laravel-data/v2/as-a-resource/transformers

**Reply by vanvanni** [reply]
Hi! Thanks for your answer. I will have a look at this :)
